sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.students.controller.updateApplicationOfferingChangeRequest.e2e-spec.ts

```diff
@@ -0,0 +1,242 @@
+        statusCode: HttpStatus.NOT_FOUND,
+        message:
+          "Application offering change not found or not in valid status to be updated.",
+        error: "Not Found",
+      });
+  });
+
+  it("Should throw a HttpStatus Not Found (404) error when an application offering change is not associated with the authenticated student.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(db.dataSource, undefined, {
+      applicationStatus: ApplicationStatus.Completed,
+    });
+    const applicationOfferingChangeRequest =
+      await saveFakeApplicationOfferingRequestChange(db, {
+        application,
+      });
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    const endpoint = `/students/application-offering-change-request/${applicationOfferingChangeRequest.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send({
+        studentConsent: true,
+        applicationOfferingChangeRequestStatus:
+          ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+      })
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        statusCode: HttpStatus.NOT_FOUND,
+        message:
+          "Application offering change not found or not in valid status to be updated.",
+        error: "Not Found",
+      });
+  });
+
+  it("Should throw a HttpStatus Not Found (404) error when an application offering change is not in a valid status to be updated.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(
+      db.dataSource,
+      { student },
+      { applicationStatus: ApplicationStatus.Completed },
+    );
+    const applicationOfferingChangeRequest =
+      await saveFakeApplicationOfferingRequestChange(
+        db,
+        {
+          application,
+        },
+        {
+          initialValues: {
+            applicationOfferingChangeRequestStatus:
+              ApplicationOfferingChangeRequestStatus.Approved,
+          },
+        },
+      );
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    const endpoint = `/students/application-offering-change-request/${applicationOfferingChangeRequest.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send({
+        studentConsent: true,
+        applicationOfferingChangeRequestStatus:
+          ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+      })
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        statusCode: HttpStatus.NOT_FOUND,
+        message:
+          "Application offering change not found or not in valid status to be updated.",
+        error: "Not Found",
+      });
+  });
+
+  it("Should throw a HttpStatus Unprocessable Entity (422) error when an invalid application offering change request is made.", async () => {
```

Comment:
I believe that we can be specific about what is invalid, in this case, the status. For instance:
"Should throw a HttpStatus Unprocessable Entity (422) error when an invalid application offering change request staus is provided."